### ISSUE TITLE
Enigma: Signed messages containing Unicode are modified when verifying the signature

### ISSUE THREAD

**Issue by ddavness** [roundcube/roundcubemail#7919]
> **Roundcube Version:** `1.4.10`, `1.4.11`

Signed messages containing Unicode characters such as ✓ will be modified - causing the underlying signatures to become invalid.

## Minimal Reproduction
Taking this dummy public key (`nobody@example.com`):
```
-----BEGIN PGP PUBLIC KEY BLOCK-----

mI0EYEEKtAEEAKPJxU5I5rIiCeosdjE4kLJPjZp4igKFw/qW5iSABna02g4Ls9p5
GAlC0R0CwVb4yDXr6pOAJG+Xx8n45yYBEsoZx9zon/e1h1u1X236xDtkP6e6TaK2
rkm59M2rk7uj05K1dDDb/BhqZLLKcWtGj7PMAawCwPedZKxeD2l+tBPdABEBAAG0
G05vYm9keSA8bm9ib2R5QGV4YW1wbGUuY29tPojOBBMBCgA4FiEERrn61bQB5bbD
tnnUrppgbs1BwAAFAmBBCrQCGwMFCwkIBwIGFQoJCAsCBBYCAwECHgECF4AACgkQ
rppgbs1BwAB8JwQAhTITGr2DToGtmW0cjBlwlbn7CQSYxqsecK1QonUsXkSH9p/B
aYCXmo2WIqXGKjVfZesDT0kmaxokYKbI1TabxuBpHxxzK5VJz+ThSC3t8BJbswve
amNUmpA1nmhPCYzvn5CfyqjpNcpSqIkoRbCqibqBuLXiGJe492cetiQjH7Q=
=1jy2
-----END PGP PUBLIC KEY BLOCK-----
```

and this signed message

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

✓ This is a unicode message (UTF-8, not ASCII)
-----BEGIN PGP SIGNATURE-----

iLMEAQEKAB0WIQRGufrVtAHltsO2edSummBuzUHAAAUCYEEMuAAKCRCummBuzUHA
ANQ2A/wM6DAraG/J4K+akkZPrceJGvikGEvfN1k2QKqRFvYgL3LZ5XTuV1K8fRNa
hObxolX260e9WonJYM0bSoX9cEJ3anfH6M+bHKGbpN4FcuM3bjPhhEy9WICmEl6v
Bu644XRJKfShlCtvikguVEG+rtm8bIlDQLzVR88grDwlcgjZhg==
=MPVT
-----END PGP SIGNATURE-----
```

(you can verify that the message above was signed by the provided key)

Enigma will hide/remove the `✓` character and complain that the signature is not valid:

![image](https://user-images.githubusercontent.com/45321798/109999657-5ab6b380-7d0a-11eb-9dde-8faa8268dccc.png)

Not signing the message will preserve the character:

![image](https://user-images.githubusercontent.com/45321798/109999887-9d788b80-7d0a-11eb-8acb-21308cee60fe.png)

**Comment by alecpl** [comment]
What client did you use to create the signed message? Could you provide a sample as a complete .eml file?

**Comment by ddavness** [comment]
This is the .eml file (I redacted emails addresses and mailservers)

```
Return-Path: <nobody@example.com>
Delivered-To: nobody@example.com
Received: from mail.example.com ([127.0.0.1])
	by mail.example.com with LMTP
	id UCM8MpINQWCzCwUAzp1sMg
	(envelope-from <nobody@example.com>)
	for <nobody@example.com>; Thu, 04 Mar 2021 16:40:50 +0000
X-Spam-Checker-Version: SpamAssassin 3.4.4 (2020-01-24) on mail.example.com
X-Spam-Level: 
X-Spam-Status: No, score=-1.1 required=5.0 tests=ALL_TRUSTED,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU autolearn=ham autolearn_force=no version=3.4.4
X-Spam-Report: 
	* -1.0 ALL_TRUSTED Passed through trusted hosts only via SMTP
	* -0.1 DKIM_VALID_AU Message has a valid DKIM or DK signature from
	*      author's domain
	*  0.1 DKIM_SIGNED Message has a DKIM or DK signature, not necessarily
	*       valid
	* -0.1 DKIM_VALID Message has at least one valid DKIM or DK signature
X-Spam-Score: -1.1
Received: from authenticated-user (mail.example.com [127.0.0.1])
	(using TLSv1.3 with cipher TLS_AES_256_GCM_SHA384 (256/256 bits)
	 key-exchange X25519 server-signature RSA-PSS (2048 bits) server-digest SHA256)
	(No client certificate requested)
	by mail.example.com (Postfix) with ESMTPSA id B76C781469
	for <nobody@example.com>; Thu,  4 Mar 2021 16:40:50 +0000 (WET)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/simple; d=nobody@example.com;
	s=mail; t=1614876050;
	bh=lglhpjfN9821F9YhoyKfsWMOGRUSD9wJbdNXlO4NGss=;
	h=Date:From:To:Subject:From;
	b=z2Pgbpw1Sj5OeL4HF0F2cK3gxQpaDIUAmDA3IM/5eU4ra8QIjXM+6RYZOgVubF+MK
	 63d+x7Uv/MNAeXBbw6ibmxHPWd1JQDJabhzGX0KxGevw6RAg+HvovMIQzVqjFt7Llt
	 JRwA56TiGnWXPULjNL+E5/OkwbHWPncGWF0WeA3iSuwGxpMoJZHxuwXwqEHQfujSUx
	 791akO8VfP9kGdOMFmxk0YyvM5xI4KJ3aXLb+YBe4FnD5oF7q0CXIpcNL4UGBVlUqW
	 qOF2Z4RlAkBX6ZHiK68fJGSx5nyCUyVddkvesISX8pXTYzrBYH1VBOzyFD+eAFVIs/
	 QPGENv5TwNL3g==
MIME-Version: 1.0
Date: Thu, 04 Mar 2021 16:40:50 +0000
From: nobody@example.com
To: nobody@example.com
Subject: Unicode Signature Test
Message-ID: <abf40480bdf75a0ce53cdaac94dddf7c@mail.example.com>
X-Sender: nobody@example.com
Content-Type: text/plain; charset=UTF-8;
 format=flowed
Content-Transfer-Encoding: 8bit

-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

✓ This is a unicode message (UTF-8, not ASCII)
-----BEGIN PGP SIGNATURE-----

iLMEAQEKAB0WIQRGufrVtAHltsO2edSummBuzUHAAAUCYEEMuAAKCRCummBuzUHA
ANQ2A/wM6DAraG/J4K+akkZPrceJGvikGEvfN1k2QKqRFvYgL3LZ5XTuV1K8fRNa
hObxolX260e9WonJYM0bSoX9cEJ3anfH6M+bHKGbpN4FcuM3bjPhhEy9WICmEl6v
Bu644XRJKfShlCtvikguVEG+rtm8bIlDQLzVR88grDwlcgjZhg==
=MPVT
-----END PGP SIGNATURE-----
```

The key and signatures were generated on GnuPG 2.2.19 on Linux.
The non-signed message was created with UTF-8 encoding via the `echo` command:

```
echo -e "\xe2\x9c\x93 This is a unicode message (UTF-8, not ASCII)" > file
```

**Comment by alecpl** [comment]
Confirmed. Since the fix for #5962 PGP message bodies are converted to ASCII. It should be done only for the encrypted part.

**Comment by alecpl** [comment]
Fixed.
